spring-session/src/main/java/org/springframework/session/data/mongo/MongoSessionRepository.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2002-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.session.data.mongo;
+
+import com.mongodb.DBObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.session.ExpiringSession;
+import org.springframework.session.FindByIndexNameSessionRepository;
+import org.springframework.session.MapSession;
+import org.springframework.session.events.SessionCreatedEvent;
+import org.springframework.session.events.SessionDeletedEvent;
+import org.springframework.session.events.SessionExpiredEvent;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Session repository implementation which stores sessions in Mongo.
+ * Uses {@link MongoSessionSerializer} to transform session objects from/to
+ * native Mongo representation ({@code DBObject}).
+ *
+ * Repository is also responsible for removing expired sessions from database.
+ * Cleanup is done every minute.
+ *
+ * @author Jakub Kubrynski
+ */
+public class MongoSessionRepository implements FindByIndexNameSessionRepository<ExpiringSession> {
```

Comment:
Please keep in mind that I am new to Mongo, so this might not work...

It seems you might be able to use  `Converter<ExpiringSession, DBObject>` and `Converter<DBObject,ExpiringSession>`. This would be implemented similar to `MongoSessionSerializer` using JDK serialization?